src/SonarScanner.MSBuild.PreProcessor/CacheProcessor.cs

```diff
@@ -0,0 +1,60 @@
+﻿/*
+ * SonarScanner for .NET
+ * Copyright (C) 2016-2022 SonarSource SA
+ * mailto: info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Threading.Tasks;
+using SonarScanner.MSBuild.Common;
+
+namespace SonarScanner.MSBuild.PreProcessor
+{
+    public class CacheProcessor
```

Comment:
In order to facilitate testing, I think we should implement an interface and use the `IPreprocessorObjectFactory` to create the right object depending on the context.

Reply:
I don't know what it means and I don't want to do it here 😄

Reply:
Let's have a talk about this. I think it's important

Reply:
Let's move it to slack. Scope of this PR is a minimal class that exists.